### ISSUE TITLE
idea: antisocial behavior vocabulary

### ISSUE THREAD

**Issue by gobengo** [swicg/activitypub-trust-and-safety#15]
ActivityStreams2 is a social data syntax and vocabulary.

It occurs to me that maybe this TF could maintain a social behavior vocabulary and e.g. an *anti*social behavior vocabulary as a part of this, and it could be useful both for community development soft skills and norm-setting as well as potentially future technical standards around moderation/safety.

I imagine this could contain good definitions of e.g. [sealioning](https://mastodon.social/@ErickaSimone/112888609931065285).

**Comment by ThisIsMissEm** [comment]
I think that would fit nicely with the idea of Labels, I mention these briefly in #1 but it's also related to #2 (categorisation is probably through the use of Labels)

For example, here's a [IFTAS](https://about.iftas.org) / [DTSP](https://dtspartnership.org/) label vocabulary for moderation purposes: https://connect.iftas.org/library/iftas-documentation/shared-vocabulary-labels/

I think these sorts of Labels can be both self-applied and applied by a moderator or third-party via Annotations (e.g., #4)

I'm actually currently writing some documentation about Labels as I've currently defined them for [FediMod Fires](https://fires.fedimod.org) (that website isn't live yet, as I'm still writing the content)

Essentially a Label is a URI that can be viewed both in the browser for a human readable definition, and also dereferenced by software to retrieve machine-readable (JSON) for the name and description of the Label, along with which label collection it is part of.

**Comment by ThisIsMissEm** [comment]
@gobengo I updated the link to the W3C Code of Conduct to point to the pull request instead of the commit: https://github.com/w3c/PWETF/pull/171 (it was merged so is part of PWETF, I guess?

**Comment by bumblefudge** [comment]
![image](https://github.com/user-attachments/assets/6253244d-a7cc-4a69-b9ff-5155d5c0d776)

**Comment by ThisIsMissEm** [comment]
@bumblefudge wanna cite your sources for that? Though also, not sure this is really the appropriate forum for a meme?

**Comment by bumblefudge** [comment]
http://users.soc.umn.edu/~uggen/Moffitt_PR_93.pdf
sorry, i thought the link was included with the screen grab. I was just googling around to see if there was a canonical taxonomy or vocab from psychology and it was something that popped up.

**Comment by ThisIsMissEm** [comment]
@bumblefudge the DTSP labels linked above are probably the best example of such, but does still have glaring omissions (e.g., nothing around animal abuse or bestiality)

i think generally these sorts of taxonomies are confined to the large social networking companies. Maybe @yoyoel or @ansavvides has more they can share.

**Comment by gobengo** [comment]
> @gobengo I updated the link to the W3C Code of Conduct to point to the pull request instead of the commit: https://github.com/w3c/PWETF/pull/171 (it was merged so is part of PWETF, I guess?

I would prefer you not edit my posts in an append-only way with an 'added by' _'. nbd though. Good find on the original PR. Good improvement.

> (it was merged so is part of PWETF, I guess?)

Yep it's here under "Unacceptable Behaviors": https://www.w3.org/policies/code-of-conduct/#unacceptablebehavior

**Comment by gobengo** [comment]
> hence leaving a comment to make it know — the original link didn't look like it had been adopted, the PR link makes that way clearer.

@ThisIsMissEm I agree it's clearer. I'm just a little bummed now I lost the link that was there. and the OP doesn't show that we both authored it.

**Comment by ThisIsMissEm** [comment]
You can click the "edited by" to get it, but: https://github.com/hober/PWETF/commit/3d93c59a0fafd6d97485252e697f5310c48fe3d5

**Comment by jfinkhaeuser** [comment]
Thanks for @bumblefudge for pointing me here.

Yes, I basically suggested a similar thing in https://socialhub.activitypub.rocks/t/moderation-related-feps/4624

A few things IMHO matter:

1. It'll be important to import/reference/adopt single labels, not only collections of labels (or whatever term we end up with here).
2. It'll be important that fedi software includes the option to import/adjust labels, and publish/reference the result.

The common goal of both comments is to avoid a single, prescriptive system.

**Comment by ThisIsMissEm** [comment]
I think that a label provider can indeed be built into Fediverse Software, but it can also stand in isolation. How specific software implements fetching & storing of labels is tbd (I actually have money in my FediMod FIRES grant from NLNet to implement labels in mastodon for domain blocks (it's a starting point)

The goal I think for now is to have a Label object that exists, and a collection of labels. A single label URI may be within the collection or on a separate route, e.g.,
- `https://provider.example/labels#violent-threat`, or, 
- `https://provider.example/labels/violent-threat`

For the later, we need the `itemOf` FEP, to allow an Object to point to the wider collection that it is a part of (I can't find the FEP right now, but I know we discussed it on SocialHub/the fedi and someone drafted a FEP for it)

It's important to note that labels could be applied to anything, so it may make sense to have a "scope" property for labels that are specific to a given type (e.g., "sockpuppet" only makes sense on an Actor, not any other Object (this would be analogous to Domain in the current AS2 vocab)

I've published some more WIP information on Labels as conceptualised for FediMod FIRES here: https://fires.fedimod.org/concepts/labels.html

**Comment by ThisIsMissEm** [comment]
@gobengo mind if I hide the more off-topic replies in this thread to try to keep it focused? (They are still visible, just collapsed)

**Comment by ThisIsMissEm** [comment]
Here's the `itemOf` pre-FEP discussion: https://socialhub.activitypub.rocks/t/pre-fep-how-to-claim-that-an-object-is-part-of-a-given-collection/4606/7

**Comment by jfinkhaeuser** [comment]
FWIW, a comment on #1 is relevant here: a vocabulary item could also be included directly in an Activity, and reference the vocabulary's entry due to having the same identifier.
